packages/app/src/game/store/sudoku-string-encoder.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable no-bitwise, no-plusplus, max-statements */
+const GRID_SIZE = 9;
```

Comment:
lets create a class instead with readonly constants similar to Solution class, move it to the history/classes folder, with its tests, add private methods to split logic where makes sense